src/SFA.DAS.Tools.Servicebus.Support.Web/Controllers/ServicebusController.cs

```diff
@@ -33,36 +41,75 @@ public class ServicebusController : Controller
             IQueryHandler<PeekQueueMessagesQuery, PeekQueueMessagesQueryResponse> peekQueueMessagesQuery,
             ICommandHandler<BulkCreateQueueMessagesCommand, BulkCreateQueueMessagesCommandResponse> bulkCreateMessagesCommand,
             ICommandHandler<SendMessagesCommand, SendMessagesCommandResponse> sendMessagesCommand,
-            IQueryHandler<GetMessagesQuery, GetMessagesQueryResponse> getMessagesQuery)
+            IQueryHandler<GetMessagesQuery, GetMessagesQueryResponse> getMessagesQuery,
+            IQueryHandler<GetMessageCountPerUserQuery, GetMessageCountPerUserQueryResponse> getMessageCountPerUserQuery,
+            IQueryHandler<GetUserSessionsQuery, GetUserSessionsQueryResponse> getUserSessionsQuery)
         {
             _logger = logger;
             _userService = userService;
             _getUserSessionQuery = getUserSessionQuery;
             _getQueuesQuery = getQueuesQuery;
             _peekQueueMessagesQuery = peekQueueMessagesQuery;
             _getMessagesQuery = getMessagesQuery;
+            _getMessageCountPerUser = getMessageCountPerUserQuery;
+            _getUserSessionsQuery = getUserSessionsQuery;
             _bulkCreateMessagesCommand = bulkCreateMessagesCommand;
             _sendMessagesCommand = sendMessagesCommand;
         }
 
         public async Task<IActionResult> Index()
         {
             var response = await _getUserSessionQuery.Handle(new GetUserSessionQuery()
-                {
-                    UserId = _userService.GetUserId()
-                });
+            {
+                UserId = _userService.GetUserId()
+            });
 
             if (response.HasSession())
             {
                 return RedirectToAction(actionName: "Index", controllerName: "MessageList");
             }            
 
-            var searchVM = new QueueViewModel
+            return View();
+        }
+
+        public async Task<IActionResult> Data()
+        {
+            var queuesResponse = await _getQueuesQuery.Handle(new GetQueuesQuery());
+
+            var messageCountResponse = await _getMessageCountPerUser.Handle(new GetMessageCountPerUserQuery());
+
+            var userSessionResponse = await _getUserSessionsQuery.Handle(new GetUserSessionsQuery());
+            var userSessions = userSessionResponse.UserSessions.ToList();
+
+            return Json(new
+            {
+                Total = queuesResponse.Queues.Count(),
+                Rows = queuesResponse.Queues.Select(q => new
+                {
+                    Id = q.Name,
+                    Name = q.Name,
+                    MessageCount = q.MessageCount,
+                    MessageCountInvestigation = messageCountResponse.QueueMessageCount.ContainsKey(q.Name) ? FormatMessageForUnderInvestigationCount(messageCountResponse.QueueMessageCount[q.Name], userSessions) : "0"
+                })
+            });
+        }
```

Comment:
I would move this into its own controller, that way we can keep the MVC and the "API" separate